R/load_fb.R

```diff
@@ -169,3 +169,59 @@ load_fb_big5_advanced_season_stats <- function(season_end_year = NA, stat_type,
 
 }
 
+#' Load pre-saved FBref match shooting event data
+#'
+#' Loading version of \code{fb_match_shooting}. Only some leagues available.
+#'
+#' @inheritParams load_match_results
+#' @importFrom purrr possibly map_dfr
+#' @importFrom cli cli_alert
+#' @importFrom dplyr filter
+#' @importFrom rlang .data
+#' @return returns a dataframe
+#' @examples
+#' \donttest{
+#' try({
+#' load_fb_match_shooting(
+#'   country = "ENG",
+#'   gender = "M",
+#'   tier = "1st"
+#' )
+#'
+#' load_fb_match_shooting(
+#'   country = c("ITA", "ESP")
+#'   gender = "M",
+#'   tier = "1st",
+#'   season_end_year = 2019
+#' )
+#' })
+#' }
+#' @export
+load_fb_match_shooting <- function(country, gender, tier, season_end_year = NA) {
+
+  ## plural because there could be multiple
+  urls <- sprintf(
+    "https://github.com/JaseZiv/worldfootballR_data/releases/download/fb_match_shooting/%s_%s_%s_match_shooting.rds",
+    country,
+    gender,
+    tier
+  )
+
+  res <- purrr::map_dfr(urls, .file_reader)
+
+  if(nrow(res) == 0) {
+    cli::cli_alert("Data not loaded. Please check parameters.")
+    return(res)
```

Comment:
return immediately so that we don't try to filter for `season_end_year` on an empty data frame